docs/_content/ontwikkelaars/algemeen/audit-trail.md

```diff
@@ -0,0 +1,248 @@
+---
+title: "Audit trail"
+date: '01-05-2019'
+weight: 50
+---
+
+Gebaseerd op: User Story [#722](https://github.com/VNG-Realisatie/gemma-zaken/issues/722)
+
+## Inleiding
+
+Consumers willen op een eenvoudige manier een audit trail kunnen opvragen zoals
+die nu veelal wordt weergeven in monolithische zaaksystemen.
+
+We hanteren voor het opstellen van een audit trail, de volgende definitie:
+
+> De audit trail bevat voldoende gegevens om achteraf te kunnen herleiden (1) 
+> welke essentiële handelingen (2) wanneer door (3a) wie of (3b) vanuit welk 
+> systeem (4) met welk resultaat zijn uitgevoerd.
+> Tot essentiële handelingen worden in ieder geval gerekend: opvoeren en 
+> afvoeren posten, statusveranderingen met wettelijke, financiële of voor de 
+> voortgang van het proces, de zaak of de klant beslissende gevolgen.
+
+Bron: [NORA](https://www.noraonline.nl/wiki/Inhoud_audit_trail)
+
+We laten de relatie met logging (in het algemeen) en doelbinding volgens de AVG
+buiten beschouwing. Hier zijn andere oplossingen voor in de maak, zoals 
+[NLx](https://nlx.io).
+
+## API implementatie
+
+### Endpoint
+
+De API wordt gerealiseerd als alleen-lezen endpoint binnen de API van het 
+component, waarbij de resource "audittrail" wordt ontsloten op het gelijknamige
+endpoint. Voor het ZRC zal dit dus zijn: `/api/v1/audittrail`.
```

Comment:
Volgens mij is een audit trail een subresource van een zaak...

Reply:
Ik had deze overwogen en eigenlijk direct afgeschoten om de volgende redenen:

1. Sub-resources maken (zaak-) overkoepelende bevragingen onmogelijk (bijv. geef alles weer wat gebruiker X heeft gedaan)
2. Een sub-resource op zaak voor de audit trail vond ik ongeschikt omdat deze ook informatie over statussen, eigenschappen, relaties met documenten, etc. terug geeft.
3. Enige meerwaarde is "convenience" omdat je niet `hoofdObject=` als filter hoeft mee te geven.

Echter, aangezien zowel jij en Alex (intern hier) hierover beginnen, heb ik het nogmaals besproken. Uitkomst, is dat het conceptueel prima past als sub-resources. Immers, alle audit regels betreffen de zaak (ook als is dat niet alleen de zaak-resource). Daarnaast maakt het de mogelijkheid om later een root-niveau endpoint toe te voegen nog gewoon open (geen irreversible move). 

Wat we echter niet gaan doen is `zaken/<id>/audittrail` en `statussen/<id>/audittrail`, etc... voor elke resource.

Reply:
ik ben voorstander van subresource omdat het echt een operatie op de zaak is, en omdat het zo eenvoudig is. Audit trail op statussen of resultaten is niet nodig.